src/System.IO.FileSystem/tests/Directory/GetFiles_str_str_so.cs

```diff
@@ -8,642 +8,321 @@
+        String testDir11Str = GetTestFileName();
+        DirectoryInfo testDir = new DirectoryInfo(TestDirectory);
+        DirectoryInfo testDir1 = testDir.CreateSubdirectory(testDir1Str);
+        testDir1.CreateSubdirectory(testDir11Str);
+
+        using (new FileInfo(Path.Combine(TestDirectory, testDir1Str, testDir11Str, GetTestFileName())).Create())
+        using (new FileInfo(Path.Combine(TestDirectory, testDir1Str, GetTestFileName())).Create())
+        {
+            Assert.Equal(2, GetFiles(TestDirectory, Path.Combine(testDir1Str, "*"), SearchOption.AllDirectories).Length);
+        }
+    }
+
+    [Fact]
+    public void InvalidSearchOption()
+    {
+        Assert.Throws<ArgumentOutOfRangeException>(() => GetFiles(".", "*", (SearchOption)100));
+        Assert.Throws<ArgumentOutOfRangeException>(() => GetFiles(".", "*", (SearchOption)(-1)));
+    }
+
+
+    [Fact]
+    public void Scenario1()
+    {
+        //Scenario 1: Vanilla - create a directory with some files and then add a couple of directories with some files and check with searchPattern *.*
+        String dirName;
+        String[] expectedFiles;
+        String[] files;
+        List<String> list;
+
+        dirName = ManageFileSystem.GetNonExistingDir(Directory.GetCurrentDirectory(), ManageFileSystem.DirPrefixName);
+        using (ManageFileSystem fileManager = new ManageFileSystem(dirName, 3, 100))
+        {
+            expectedFiles = fileManager.GetAllFiles();
+            list = new List<String>(expectedFiles);
+            files = Directory.GetFiles(dirName, "*.*", SearchOption.AllDirectories);
+            Eval(files.Length, list.Count, "Err_415mbz! wrong file count.");
+            for (int i = 0; i < files.Length; i++)
             {
-                dirName = ManageFileSystem.GetNonExistingDir(Directory.GetCurrentDirectory(), ManageFileSystem.DirPrefixName);
-                using (ManageFileSystem fileManager = new ManageFileSystem(dirName, 1, 10))
-                {
-                    expectedFiles = fileManager.GetFiles(dirName, 0);
-                    list = new List<String>(expectedFiles);
-                    files = Directory.GetFiles(dirName, "*.*", SearchOption.TopDirectoryOnly);
-                    Eval(files.Length, list.Count, "Err_3947g! wrong file count.");
-                    for (int i = 0; i < files.Length; i++)
-                    {
-                        if (Eval(list.Contains(files[i]), "Err_582bmw! unexpected file found: {0}", files[i]))
-                            list.Remove(files[i]);
-                    }
-                    if (!Eval(list.Count == 0, "Err_891vut! {0} expected files not found.", list.Count))
-                    {
-                        foreach (String fileName in list)
-                            Console.WriteLine(fileName);
-                    }
-                }
+                if (Eval(list.Contains(files[i]), "Err_287kkm! unexpected file found: {0}", files[i]))
```

Comment:
If they do have some value, I couldn't find it :) They should be assertions. They're holdovers from before xunit was being used for testing.